smartsim/ray/raycluster.py

```diff
@@ -0,0 +1,378 @@
+    and the first host is used to launch the head node.
+    
+    :param name: The name of the entity.
+    :type name: str
+    :param path: path to output, error, and configuration files
+    :type path: str
+    :param ray_port: Port at which the head node will be running.
+    :type ray_port: int
+    :param ray_num_cpus: Number of CPUs used by Ray
+    :type ray_num_cpus: int
+    :param workers: Number of workers (the total size of the cluster is ``workers``+1).
+    :type workers: int
+    :param run_args: Arguments to pass to launcher to specify details such as partition, time, and so on.
+    :type run_args: dict[str,str]
+    :param launcher: Name of launcher to use for starting the cluster.
+    :type launcher: str
+    :param alloc: ID of allocation to run on, only used if launcher is Slurm and allocation is
+                  obtained with ``ray.slurm.get_allocation``
+    :type alloc: int
+    :param batch: Whether cluster should be launched as batch file, ignored when ``launcher`` is `local`
+    :type batch: bool
+    :param time: The walltime the cluster will be running for
+    :type time: str
+    """
+
+    def __init__(self,
+                 name,
+                 path=".",
+                 ray_port=6780,
+                 ray_num_cpus=12,
+                 workers=0,
+                 run_args={},
+                 launcher='local',
+                 batch=False,
+                 time="01:00:00",
+                 alloc=None):
+        if launcher=='local' and workers>0:
+            raise SSUnsupportedError('Cannot launch a local Ray cluster with more than one node.'
+                                     'Set workers to 0 and restart.')
+        self._workers = workers
+        self._ray_port = ray_port
+        self._ray_password = uuid.uuid4()
+        self._launcher = launcher.lower()
+        self._run_args = run_args
+        self._ray_num_cpus = ray_num_cpus
+        self.head_model = None
+        self.worker_model = None
+        self.alloc = None
+        self.batch_settings = None
+        self._alloc = alloc
+        self._batch = launcher!='local' and batch
+        self._hosts = None
+        self._time = time
+        super().__init__(name=name, path=path)
+    
+    @property
+    def batch(self):
+        return self._batch   
+
+    def _update_worker_model(self):
+        self.worker_model.update_run_settings()
+
+    def _initialize_entities(self):
+        self.head_model = RayHead(name="head", params=None,
+                                  path=os.path.join(self.path, 'head'),
+                                  ray_password = self._ray_password,
+                                  ray_port=self._ray_port,
+                                  launcher=self._launcher,
+                                  run_args=self._run_args,
+                                  ray_num_cpus=self._ray_num_cpus,
+                                  alloc=self._alloc,
+                                  batch=self.batch,
```

Comment:
Can a user specify a queue or a partition for batch workloads? I see ``run_args`` but no method for parameter or ``batch_args``

Reply:
Mh, right. I might have mixed up the two.

Reply:
Fixed